crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -224,9 +232,87 @@ impl CumulocityConverter {
             c8y_endpoint,
             mqtt_schema: MqttSchema::default(),
             entity_store,
+            external_id_map,
+            reverse_external_id_map,
         })
     }
 
+    fn try_convert_entity_registration(
+        &mut self,
+        entity_topic_id: &EntityTopicId,
+        input: &EntityRegistrationMessage,
+    ) -> Result<Message, ConversionError> {
+        // Parse the optional fields
+        let display_name = input.payload.get("name").and_then(|v| v.as_str());
+        let display_type = input.payload.get("type").and_then(|v| v.as_str());
+
+        // Generate the c8y external id and cache it
+        let external_id = input
+            .entity_id
+            .as_ref()
+            .map(|v| v.to_string())
+            .unwrap_or_else(|| self.map_to_c8y_external_id(entity_topic_id));
+        self.external_id_map
+            .insert(entity_topic_id.clone(), external_id.clone());
+        self.reverse_external_id_map
+            .insert(external_id.clone(), entity_topic_id.clone());
+
+        match input.r#type {
+            EntityType::MainDevice => Err(ConversionError::MainDeviceRegistrationNotSupported),
+            EntityType::ChildDevice => {
+                let ancestors = self.entity_store.ancestors(entity_topic_id)?;
+                let ancestors_external_ids = self.ancestors_external_ids(&ancestors);
```

Comment:
Will create a new helper function over this piece of code that'll return the target entity's C8y topic (e.g: `c8y/s/us/entity-1/entity-2/entity-3`) directly from the `EntityTopicId`.